File: android/src/main/java/com/witsystem/top/flutterwitsystem/add/ble/AddBleDeviceCall.java
package com.witsystem.top.flutterwitsystem.add.ble;

/**
 * 添加设备的回调
 */
public interface AddBleDeviceCall {


    /**
     * 扫描到设备的回调
     *
     * @param deviceId
     * @param rssi
     */
    void scanDevice(String deviceId, int rssi);


    /**
     * 添加进度回调，不是百分比，只是当前进行到那一笔的回调码
     *
     * @param code
     */
    void addProcess(String deviceId,int code);


    /**
     * 异常信息的回调
     */
    void error(String deviceId,String err, int code);


    /**
     * 添加成功的回调
     */
    void addSuccess(String deviceId, int code);


}

File: ios/Classes/add/ble/AddBleDeviceCall.swift
//
// Created by yyjjls on 2020/7/11.
//

import Foundation

/**
 * 添加设备的回调
 */
protocol AddBleDeviceCall {

    /**
        * 扫描到设备的回调
        *
        */
    func scanDevice(deviceId: String?, rssi: Int);


    /**
     * 添加进度回调，不是百分比，只是当前进行到那一笔的回调码
     */
    func addProcess(deviceId: String?, code: Int);


    /**
     * 异常信息的回调
     */
    func error(deviceId: String?, err: String, code: Int);


    /**
     * 添加成功的回调
     */
    func addSuccess(deviceId: String?, code: Int);

}
